compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -283,6 +283,14 @@ fn check_main_fn_ty(tcx: TyCtxt<'_>, main_def_id: DefId) {
         error = true;
     }
 
+    if !tcx.codegen_fn_attrs(main_def_id).target_features.is_empty()
+        && !tcx.sess.target.is_like_wasm
+        && !tcx.sess.opts.actually_rustdoc
+    {
```

Comment:
is there any way you can avoid this actually_rustdoc check? these should not be added 😅
maybe overwrite the codegen_fn_attrs query in rustdoc? Or why exactly was this needed in the first place?

Reply:
I basically copied-pasted the condition that allows `#[target_feature]` on safe functions (see the rationale for `actually_rustdoc` at the bottom of the comment): https://github.com/rust-lang/rust/blob/104f4300cfddbd956e32820ef202a732f06ec848/compiler/rustc_codegen_ssa/src/codegen_attrs.rs#L227-L244

Reply:
I guess it's okay then for now, we just need to make sure that they are both fixed at the same time. Could you add a comment to the one you added?